control-plane-operator/controllers/hostedcontrolplane/oauth/idp_convert.go

```diff
@@ -0,0 +1,570 @@
+package oauth
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/cache"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/klog/v2"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	configv1 "github.com/openshift/api/config/v1"
+	osinv1 "github.com/openshift/api/osin/v1"
+
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/util"
+)
+
+const (
+	IDPVolumePathPrefix = "/etc/oauth/idp"
+)
+
+var (
+	osinv1Scheme     = runtime.NewScheme()
+	osinv1Codecs     = serializer.NewCodecFactory(osinv1Scheme)
+	osinv1Encoder    = osinv1Codecs.LegacyCodec(osinv1.GroupVersion)
+	osinv1Serializer = jsonserializer.NewSerializerWithOptions(
+		jsonserializer.DefaultMetaFactory, osinv1Scheme, osinv1Scheme,
+		jsonserializer.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
```

Comment:
I think all of this stuff could be removed by updating the [global scheme](https://github.com/openshift/hypershift/blob/main/api/scheme.go#L38) and using the global serializer.